common/src/main/java/ch/admin/bag/covidcertificate/common/qr/QrScanFragment.kt

```diff
@@ -147,34 +153,36 @@ abstract class QrScanFragment : Fragment() {
 				.setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
 				.build()
 				.also { imageAnalysis ->
-					imageAnalysis.setAnalyzer(mainExecutor, QRCodeAnalyzer { decodeCertificateState: DecodeCertificateState ->
-						when (decodeCertificateState) {
-							is DecodeCertificateState.ERROR -> {
-								handleInvalidQRCodeExceptions(decodeCertificateState.error)
-							}
-							DecodeCertificateState.SCANNING -> {
-								view?.post { updateQrCodeScannerState(QrScannerState.NO_CODE_FOUND) }
-							}
-							is DecodeCertificateState.SUCCESS -> {
-								val qrCodeData = decodeCertificateState.qrCode
-								qrCodeData?.let {
-									decodeQrCodeData(
-										it,
-										onDecodeSuccess = {
-											// Once successfully decoded, clear the analyzer from stopping more frames being
-											// analyzed and possibly decoded successfully
-											imageAnalysis.clearAnalyzer()
-
-											view?.post { updateQrCodeScannerState(QrScannerState.VALID) }
-										},
-										onDecodeError = { error ->
-											view?.post { handleInvalidQRCodeExceptions(error) }
-										}
-									)
+					imageAnalysis.setAnalyzer(
+						mainExecutor,
+						QRCodeMLKitAndXingAnalyzer(lifecycle.coroutineScope) { decodeCertificateState: DecodeCertificateState ->
```

Comment:
It's better to use `viewLifecycleOwner.lifecycleScope` instead of `lifecycle.coroutineScope` so that it's scoped to the view lifecycle and not the fragment lifecycle